File: sources/go/binary_gap.go
package main

import ("fmt",
     "math/Mod",
	 "math/Floor",
)

func  main() {
	fmt.Println(execute(654345))
}

func execute(value int) (result int) {
	//fmt.Println(value)
	for  mmod:=math.Mod(value, 2); value > 0 && mmod == 0;  {
            value := value / 2
	}

	currentGap := 0;
	var result int 
	maxGap := 0;
	for ;value > 0; {
		remainder := value % 2;
		if remainder == 0 {
			currentGap++;
		} else if currentGap != 0 {
			maxGap := Math.max(currentGap, maxGap);
			currentGap := 0
		}
		value := value / 2
	}
	result := maxGap
	return result;
}

File: sources/py/binary_gap.py
""" BinaryGap algorithm """

#!/usr/bin/env python3
# pylint: skip-file
from operator import mod

def execute(value):
    while value > 0 and (mod(value, 2) == 0):
        value = value / 2

    currentGap = 0
    maxGap = 0
    while int(value) > 0:
        remainder = int(mod(value, 2))
        if remainder == 0:
            currentGap += 1
        elif currentGap != 0:
            maxGap = max(currentGap, maxGap)
            currentGap = 0
        value = value / 2
    return maxGap

if __name__ == "__main__":
    result = execute(654345)

    print("Binary Gap: {result} ".format(
        result=str(result)))
